.github/build.sh

```diff
@@ -231,4 +231,4 @@ fi
 cmake --build . --target test-aws-iot-device-client
 
 ### Run Tests ###
-./test/test-aws-iot-device-client
+env AWS_CRT_MEMORY_TRACING=1 ./test/test-aws-iot-device-client
```

Comment:
Why are we not adding tests for `AWS_CRT_MEMORY_TRACING=2 OR 0`?

Reply:
You are referring to tests that the configuration value is interpreted from environment correctly rather than tests of the tracing capability right? I agree with the former and will add.